lib/app/widgets/scaffold.dart

```diff
@@ -8,37 +8,154 @@ class FancyScaffold extends StatelessWidget {
     Key key,
     @required this.appBar,
     @required this.sliver,
+    this.floatingActionButton,
     this.omitHorizontalPadding = false,
   })  : assert(appBar != null),
         assert(sliver != null),
+        assert(omitHorizontalPadding != null),
         super(key: key);
 
   final Widget appBar;
   final Widget sliver;
+  final Widget floatingActionButton;
   final bool omitHorizontalPadding;
 
   @override
   Widget build(BuildContext context) {
-    final padding = context.mediaQuery.padding;
-
     return Scaffold(
       body: CustomScrollView(
         slivers: <Widget>[
           appBar,
           SliverPadding(
-            padding: EdgeInsets.fromLTRB(
-              max(16, padding.left),
-              8,
-              max(16, padding.right),
-              16,
-            ),
+            padding: _paddingForScaffold(context, omitHorizontalPadding),
             sliver: MediaQuery.removePadding(
+              removeLeft: true,
+              removeRight: true,
               context: context,
               child: sliver,
             ),
           ),
         ],
       ),
+      floatingActionButton: floatingActionButton,
     );
   }
 }
+
+class FancyTabbedScaffold extends StatelessWidget {
+  const FancyTabbedScaffold({
+    Key key,
+    @required this.appBarBuilder,
+    this.controller,
+    @required this.tabs,
+    this.omitHorizontalPadding = false,
+  })  : assert(appBarBuilder != null),
+        assert(tabs != null),
+        super(key: key);
+
+  final Widget Function(bool) appBarBuilder;
```

Comment:
Positional `bool` arguments are hard to understand. Either define a typedef for this function type or name the `bool` argument.